Tooling - Update RuboCop config to not allow shorthand hash syntax

iangmaia opened the pull request:
I noticed in a review that Ruby method calls in the WPiOS project were still allowing to omit parameter values, a style we decided (see paaHJt-5yi-p2) not to follow.

This PR adds the RuboCop configuration below and fixes the corresponding code.
```yaml
Style/HashSyntax:
  EnforcedShorthandSyntax: never
```

iangmaia committed:
Update RuboCop config to not allow shorthand hash syntax

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Track `bundle exec pod install` result

The `Podfile` changed in the previous commit,
b70304f4f7d50301b4e37b395502fe771cb87a48 resulting in the need to track
the computed hash change in the `Podfile.lock`.

mokagio reviewed (approved):
Thanks @iangmaia ! I've been thinking about doing this every time I wrote a line in the previous style, but never got to actually implementing the change.

I pushed 59f91fa146 which should unblock CI. I'll enable auto-merge to get this in sooner rather than later.
